__tests__/index.ts

```diff
@@ -80,6 +80,106 @@ describe('test suites', () => {
   );
 });
 
+describe('issue 58 - key and value being identical', () => {
+  it('should handle duplicate but equal `Map` entries', () => {
+    const mapA = new Map<any, any>([
+      [{ b: 'c' }, 2],
+      [{ b: 'c' }, 2],
+    ]);
+    const mapB = new Map<any, any>([
+      [{ b: 'c' }, 2],
+      [{ b: 'c' }, 2],
+    ]);
+
+    const result = deepEqual(mapA, mapB);
+
+    expect(result).toBe(true);
+  });
+
+  it('should handle the first `Map` entry being unequal', () => {
+    const mapA = new Map<any, any>([
+      [{ b: 'c' }, 2],
+      [{ b: 'c' }, 2],
+    ]);
+    const mapB = new Map<any, any>([
+      [{ b: 'c' }, 2],
+      ['foo', 'different'],
+    ]);
+
+    const result = deepEqual(mapA, mapB);
```

Comment:
For "first", "last" and "intermediary"  test we could create "reversed" tests with `mapA` and `mapB` switched in the call of the function. 
Since we have two loops in areMapsEqual, the order is important. For example, without fixing areMapsEqual and just changing the order in which we iterate (switching a and b) the tests will pass.

Reply:
Good call, I like the concept. I'll update.